blockchain/chainio.go

```diff
@@ -1167,6 +1167,44 @@ func (b *BlockChain) initChainState() error {
 			return err
 		}
 
+		// Load all of the headers from the data for the known best
+		// chain and construct the block index accordingly.  Since the
+		// number of nodes are already known, perform a single alloc
+		// for them versus a whole bunch of little ones to reduce
+		// pressure on the GC.
```

Comment:
Nice memory optimization! Packing them the raw values into a single slice (rather than using a slice of pointers) will also indirectly improve performance as accesses will have better cache locality.